.github/workflows/build.yml

```diff
@@ -10,6 +10,9 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment:
Hey Jax, thanks for doing this! Can you tell me how this allows CI to run as expected on forks? Just high level explanation would be fine!

Reply:
### Ref Line
`github.event` references where we want to run CI ([context docs](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)) in our case we want to run CI on the `pull_request`. For PRs we want to run CI on the pull request's head branch name `head.ref` vs the base branch name ([PR workflow events docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)). So this line `github.event.pull_request.head.ref` is essentially saying: run CI on the branch the PR was opened from.

### Repository Line
`github.event.pull_request.head` is the same as above so it's saying for the repository use the PRs head vs main. Then `repo.full_name` pulls the full path of the repo the fork is on vs attempting to pull the branch name from our path where it wouldn't exist ([reference docs here](https://docs.github.com/en/webhooks/webhook-events-and-payloads#pullrequestevent)).

Reply:
Thank you Jax!